contrib/epee/src/mlocker.cpp

```diff
@@ -59,7 +65,7 @@ static void do_lock(void *ptr, size_t len)
 {
 #if defined HAVE_MLOCK
   int ret = mlock(ptr, len);
-  if (ret < 0)
+  if (ret < 0 && !previously_failed.exchange(true))
     MERROR("Error locking page at " << ptr << ": " << strerror(errno));
```

Comment:
Could you please add something like "any subsequent failure will be silenced" or similar ?